lib/internal/Magento/Framework/App/Response/Http.php

```diff
@@ -131,7 +133,7 @@ public function sendVary()
      */
     public function setPublicHeaders($ttl)
     {
-        if ($ttl < 0 || !preg_match('/^[0-9]+$/', $ttl)) {
+        if ($ttl < 0 || !$ttl) {
```

Comment:
Hi @mrtuvn, any info of why the following logic has been changed?

Reply:
After add strict type i have found error here when second argument of preg_match required type string. But we passed int value

Reply:
I think the main reason of why it was failing, is the following test that passes a `null` value here https://github.com/magento/magento2/blob/45f27d64cbb475a4af4d27e14d312653823083e1/lib/internal/Magento/Framework/App/Test/Unit/Response/HttpTest.php#L231

But preg_match's 2nd arguments should be a string value.

But now, we allow passing string values as parameter, as we still don't have any restrictions, which shouldn't happen:
- asd
- test123
- =1
- etc

We should avoid that, and should only have int values.